tests/Parse/ParsePolygonTest.php

```diff
@@ -0,0 +1,165 @@
+
+        $obj = ParseObject::create('TestObject');
+        $obj->set('polygon', $polygon);
+        $obj->save();
+
+        // Query by open points
+        $query = new ParseQuery('TestObject');
+        $query->equalTo('polygon', $polygon);
+
+        $results = $query->find();
+        $actualPolygon = $results[0]->get('polygon');
+
+        $this->assertEquals(1, count($results));
+        $this->assertEquals($closedPoints, $actualPolygon->getCoordinates());
+
+        // Query by closed points
+        $polygon = new ParsePolygon($closedPoints);
+        $query = new ParseQuery('TestObject');
+        $query->equalTo('polygon', $polygon);
+
+        $results = $query->find();
+        $actualPolygon = $results[0]->get('polygon');
+
+        $this->assertEquals(1, count($results));
+        $this->assertEquals($closedPoints, $actualPolygon->getCoordinates());
+    }
+
+    public function testPolygonWithGeoPoints()
+    {
+        $p1 = new ParseGeoPoint(0, 0);
+        $p2 = new ParseGeoPoint(0, 1);
+        $p3 = new ParseGeoPoint(1, 1);
+        $p4 = new ParseGeoPoint(1, 0);
+        $p5 = new ParseGeoPoint(0, 0);
+
+        $points = [$p1, $p2, $p3, $p4];
+        $openPoints = [[0,0],[0,1],[1,1],[1,0]];
+        $closedPoints = [[0,0],[0,1],[1,1],[1,0],[0,0]];
+        $polygon = new ParsePolygon($points);
+
+        $obj = ParseObject::create('TestObject');
+        $obj->set('polygon', $polygon);
+        $obj->save();
+
+        // Query by open points
+        $query = new ParseQuery('TestObject');
+        $query->equalTo('polygon', $polygon);
+
+        $results = $query->find();
+        $actualPolygon = $results[0]->get('polygon');
+
+        $this->assertEquals(1, count($results));
+        $this->assertEquals($closedPoints, $actualPolygon->getCoordinates());
+
+        // Query by closed points
+        $polygon = new ParsePolygon($closedPoints);
+        $query = new ParseQuery('TestObject');
+        $query->equalTo('polygon', $polygon);
+
+        $results = $query->find();
+        $actualPolygon = $results[0]->get('polygon');
+
+        $this->assertEquals(1, count($results));
+        $this->assertEquals($closedPoints, $actualPolygon->getCoordinates());
+    }
+
+    public function testPolygonMinimum()
+    {
+        $this->setExpectedException(
+            '\Parse\ParseException',
+            'Polygon must have at least 3 points'
+        );
+        $polygon = new ParsePolygon([[0,0]]);
+        $obj = ParseObject::create('TestObject');
+        $obj->set('polygon', $polygon);
+        $obj->save();
+    }
+
+    public function testPolygonInvalidInput()
+    {
+        $this->setExpectedException(
+            '\Parse\ParseException',
+            'Polygon must have at least 3 points'
+        );
+        $polygon = new ParsePolygon(1234);
+        $obj = ParseObject::create('TestObject');
+        $obj->set('polygon', $polygon);
+        $obj->save();
+    }
+
```

Comment:
Assuming you add the aforementioned exceptions to further scrutinize the input to `ParsePolygon`, you would have to add a couple cases here for firing the other conditions.